ner_processing/message.py

```diff
@@ -0,0 +1,51 @@
+from typing import List, Dict, Any
+
+from PyQt6.QtCore import QDateTime
+
+from ner_processing.data import Data
+from ner_processing.master_mapping import DECODE_DATA, MESSAGE_IDS
+
+
+class MessageFormatException(Exception):
+    """
+    A class to represent exceptions related to invalid message formats.
+    """
+
+    def __init__(self, message: str, status: int = 0):
+        self.message = message
+        self.status = status
+
+
+class Message:
+    """
+    Wrapper class for an individual message.
+    """
+
+    def __init__(self, timestamp: QDateTime, id: int, data: List[int]):
+        self.timestamp = timestamp
+        self.id = id
+        self.data = data
+
+    def __str__(self):
+        """
+        Overrides the string representation of the class.
+        """
+        return f"[{self.timestamp.toString('yyyy-MM-ddTHH:mm:ss.zzzZ')}] {self.id} - {self.data}"
+
+    def decode(self) -> List[Data]:
+        """
+        Processes this message's data into a list of data points.
+        """
+        return self.decodeMessage(self.timestamp, self.id, self.data)
+
+    @staticmethod
+    def decodeMessage(timestamp: QDateTime, id: int, data: List[int]) -> List[Data]:
```

Comment:
Damn this is some fancy python syntax

Reply:
I found it much more readable to add type hints, especially with all these pyqt classes floating around. Hoping it decreases errors when we have more people developing on it.